### PR TITLE
boards/nucleo: factorize code of the different board types (32 pins, 144 pins)

### PR DESCRIPTION
**Pull request by aabadie**
This PR also adds/fixes Arduino support for nucleo-32 boards.

The serial configuration is shared between the different nucleo board types: 32, 64 and 144 pins.

We can even more factorize : move cpp feature to nucleo-common and share it with nucleo32 and nucleo144. The inheritance tree could also be improved with a common folder for all nucleo types, then a common for each board type (32, 64, 144), this is not what is done here but it can be changed if it makes more sense to the reviewers.

### CODE REVIEW

File: boards/nucleo144-common/include/gpio_params.h [hunk 1]
```diff
@@ -1,21 +1,19 @@
 /*
- * Copyright (C) 2017 Inria
- *               2017 OTA keys
+ * Copyright (C) Inria 2017
```
Comment (lebrush) [inline boards/nucleo144-common/include/gpio_params.h]: Copyright date switch

File: boards/nucleo32-common/include/gpio_params.h [hunk 2]
```diff
@@ -1,13 +1,13 @@
 /*
- * Copyright (C) 2017   Inria
+ * Copyright (C) Inria 2017
```
Comment (lebrush) [inline boards/nucleo32-common/include/gpio_params.h]: same here

**Comment by lebrush** [comment]
I like this refactoring :+1: and codewise looks good. 

Are you planning to include the 64 pin boards (there are many more of these more than of 32/144)?

File: boards/nucleo144-f207/Makefile.include [hunk 3]
```diff
@@ -2,5 +2,8 @@
 export CPU = stm32f2
 export CPU_MODEL = stm32f207zg
 
-# load the common Makefile.include for Nucleo boards
-include $(RIOTBOARD)/nucleo-common/Makefile.include
+# include nucleo common serial configuration
+include $(RIOTBOARD)/nucleo-common/Makefile.include.serial
```
Comment (aabadie) [inline boards/nucleo144-f207/Makefile.include]: looking at my changes again, this line is not required as it's already include by indirectly by the line above. I'll update this for others.

File: boards/nucleo144-f207/Makefile.include [hunk 4]
```diff
@@ -2,5 +2,8 @@
...
+
+# load the common Makefile.include for Nucleo-32 boards
```
Comment (aabadie) [inline boards/nucleo144-f207/Makefile.include]: nucleo-144 (auto-review ;) )

**Comment by aabadie** [comment]
@lebrush thanks for having a look.

> Are you planning to include the 64 pin boards (there are many more of these more than of 32/144)?

In fact, the 64 pins are taken as reference as they are many more. So I think I'll keep it as it is in this PR.
I'll fix the copyright issue and a few other inconsistencies I just found.

**Comment by aabadie** [comment]
@lebrush I directly rebased and force push.

File: boards/nucleo144-common/include/gpio_params.h [hunk 5]
```diff
@@ -34,10 +32,25 @@ extern "C" {
...
+    },
+    {
+        .name = "LD3(red)",
+        .pin = LED2_PIN,
+        .mode = GPIO_OUT
+    },
+    {
+        .name = "B1(User button)",
+        .pin = BTN_B1_PIN,
+        .mode = GPIO_IN_PU
```
Comment (vincent-d) [inline boards/nucleo144-common/include/gpio_params.h]: On nucleo144, ti pin is pulled-down externally, so it should be either `GPIO_IN` or `GPIO_IN_PD`

Comment (aabadie) [inline boards/nucleo144-common/include/gpio_params.h]: Thanks ! I changed it to `GPIO_IN_PD`

**Comment by aabadie** [comment]
Looks like I broke a few things with gpio params, will fix.

**Comment by aabadie** [comment]
Rebased on master with #6714 inside. Related boards needs to be tested:
* Nucleo 32
  - [ ] nucleo32-f031
  - [ ] nucleo32-f042
  - [ ] nucleo32-f303
* Nucleo 144
  - [ ] nucleo144-f207
  - [ ] nucleo144-f413

**Comment by aabadie** [comment]
Tested nucleo32-f042 and nucleo-f303 : works

**Comment by vincent-d** [comment]
I'll test the nucleo144-f207 and nucleo144-f413 by the end of the week if you don't have them.

**Comment by aabadie** [comment]
I have the 207 but not the 413, thanks

**Comment by aabadie** [comment]
nucleo144-f207 works

**Comment by vincent-d** [comment]
examples/default works on nucleo144-f413 (tested saul and rtc)

**Comment by aabadie** [comment]
Works also with nucleo32-f031 and nucleo32-l031. If someone (@vincent-d, @lebrush) could ACK that would be great.

**Comment by vincent-d** [comment]
And go!

**Comment by aabadie** [comment]
@vincent-d thanks ! but you forgot to wait for Murdock :)

**Comment by vincent-d** [comment]
@aabadie I saw only green on my screen...either a github lag, or my eyes. Sorry!

**Comment by aabadie** [comment]
np @vincent-d, it's just that Murdock is our "official" CI. Check that "Ready for CI build" label is set before pressing the green button ;)

**Pull request merged**